src/ecdh.rs

```diff
@@ -47,6 +47,12 @@ const SHARED_SECRET_SIZE: usize = constants::SECRET_KEY_SIZE;
 pub struct SharedSecret([u8; SHARED_SECRET_SIZE]);
 impl_display_secret!(SharedSecret);
 
+impl From<[u8; SHARED_SECRET_SIZE]> for SharedSecret {
+    fn from(arr: [u8; SHARED_SECRET_SIZE]) -> Self {
+        SharedSecret(arr)
+    }
+}
```

Comment:
I'm personally not keen on adding `impl From<Inner> for Newtype` in case there is `struct Newtype(Inner);` It encourages just calling `into()` which hides the newtype.

I suggest `from_bytes(bytes: [u8; SHARED_SECRET_SIZE]) -> Self` method instead.

Reply:
I don't have a strong preference here. I'm not sure what you mean by "hiding" .. a call to `.into` is pretty explicit.

Reply:
It's not clear which type is converted into which until you review a bigger chunk of code.

Reply:
Done. I'm not very familiar with `#[inline]` usage, let me know if this is needed here as well.

Reply:
Another data point: we have `secret_bytes() -> [u8; SHARED_SECRET_SIZE]` already which is slightly strangely named, the reason is that its the same method name used for various keys, especially `KeyPair` (for which it makes most sense). In light of this I'd find `from_bytes` the least surprising name so as kind of match with `secret_bytes`.

Reply:
`#[inline]` could be useful/nice but not super-important. Especially if one uses LTO.

Reply:
@Kixunil @apoelstra @dspicher 

I think it is a poor decision to avoid implementing traits from the standard library which are even part of the [standard prelude](https://doc.rust-lang.org/std/prelude/index.html#prelude-contents). The `From` trait defines a standard interface that allows for better composability, e.g. you can define functions like this

```rust
fn process<T: Into<Foo>>(t: T)
```

and it will compile for all types that can be converted into `Foo`. Also, you can always opt-in for doing the conversion explicitly:

```rust
let bar: Bar = foo.into()
```

I really don't understand why we need to segue here out of the standard rust interface for doing conversions.